lib/db.js

```diff
@@ -302,6 +315,75 @@ class LimitDBRedis extends EventEmitter {
       });
   }
 
+  buildStatusResponse(bucket, prefixLength) {
+    return (key, lastDrip, remaining) => {
+      const bucketKeyConfig = this.bucketKeyConfig(bucket, key);
+      return {
+        key: key.substring(prefixLength),
+        limit: bucketKeyConfig.size,
+        reset: this.calculateReset(bucketKeyConfig, remaining),
+        remaining
+      };
+    };
+  }
+
+  /**
+   * Find all buckets that match the key.
+   *
+   * @param {statusParams} params
+   * @param {function(Error, statusResult)} [callback]
+   */
+  status(params, callback) {
+    const valError = this.validateParams(params);
+    if (valError) {
+      return setImmediate(callback, valError);
+    }
+    const bucket = this.buckets[params.type];
+    let prefix = `${params.type}:${params.key}*`;
+    let prefixLength = `${params.type}:`.length;
+
+    // Calling nodes directly does not use the global prefix
+    if (this.prefix) {
+      prefix = `${this.prefix}${prefix}`;
+      prefixLength = `${this.prefix}${params.type}:`.length
+    }
+
+    const count = params.count || 50;
+    const transform = this.buildStatusResponse(bucket, prefixLength);
+
+    const nodes = this.redis.nodes ? this.redis.nodes('slave') : [this.redis];
+    const errors = [];
+    async.map(nodes, (node, cb) => {
+      node.bucketStatus(prefix, count, (err, results) => {
+        if (err) {
+          errors.push(err);
+          this.emit('error', err);
```

Comment:
I know I told you about emitting, but that was before the array of errors, what do you think of keeping just the array now that we are not blind anymore?